modules/service/src/main/resources/db/migration/V0995__imaging_science_requirements.sql

```diff
@@ -1,93 +1,102 @@
--- the exposure time mode is shared between imaging and spectroscopy
-ALTER TABLE t_observation RENAME COLUMN c_spec_exp_time_mode      TO c_exp_time_mode;
-ALTER TABLE t_observation RENAME COLUMN c_spec_signal_to_noise    TO c_etm_signal_to_noise;
-ALTER TABLE t_observation RENAME COLUMN c_spec_signal_to_noise_at TO c_etm_signal_to_noise_at;
-ALTER TABLE t_observation RENAME COLUMN c_spec_exp_time           TO c_etm_exp_time;
-ALTER TABLE t_observation RENAME COLUMN c_spec_exp_count          TO c_etm_exp_count;
+-- Add constraint to ensure mutual exclusivity between imaging and spectroscopy fields
+ALTER TABLE t_observation ADD CONSTRAINT check_science_mode_exclusivity
+CHECK (
+  -- Either all spectroscopy fields are NULL (num_nulls = 6)
+  num_nulls(c_spec_wavelength, c_spec_resolution, c_spec_wavelength_coverage,
+            c_spec_focal_plane, c_spec_focal_plane_angle, c_spec_capability) = 6
+  OR
+  -- Or all imaging fields are NULL (num_nulls = 4)
+  num_nulls(c_img_minimum_fov, c_img_narrow_filters,
+            c_img_broad_filters, c_img_combined_filters) = 4
+  -- This allows all NULL
+);
 
--- Add check constraint to ensure c_science_mode is consistent with spectroscopy and imaging fields
-ALTER TABLE t_observation
-  ADD CONSTRAINT science_mode_consistency CHECK (
-    CASE c_science_mode
-      -- If spectroscopy mode, all imaging fields must be NULL
-      WHEN 'spectroscopy' THEN
-        c_img_minimum_fov IS NULL AND
-        c_img_narrow_filters IS NULL AND
-        c_img_broad_filters IS NULL AND
-        c_img_combined_filters IS NULL
-      -- If imaging mode, all spectroscopy fields must be NULL
-      WHEN 'imaging' THEN
-        c_spec_wavelength IS NULL AND
-        c_spec_resolution IS NULL AND
-        c_spec_wavelength_coverage IS NULL AND
-        c_spec_focal_plane IS NULL AND
-        c_spec_focal_plane_angle IS NULL AND
-        c_spec_capability IS NULL
-      -- If science mode is NULL, all science requirement fields must be NULL
-      ELSE
-        c_spec_wavelength IS NULL AND
-        c_spec_resolution IS NULL AND
-        c_spec_wavelength_coverage IS NULL AND
-        c_spec_focal_plane IS NULL AND
-        c_spec_focal_plane_angle IS NULL AND
-        c_spec_capability IS NULL AND
-        c_img_minimum_fov IS NULL AND
-        c_img_narrow_filters IS NULL AND
-        c_img_broad_filters IS NULL AND
-        c_img_combined_filters IS NULL
-    END
-  );
+-- Create function to check if spectroscopy mode is active
+CREATE OR REPLACE FUNCTION has_spectroscopy_mode(
+  spec_wavelength          d_wavelength_pm,
+  spec_resolution          integer,
+  spec_wavelength_coverage d_wavelength_pm,
+  spec_focal_plane         d_tag,
+  spec_focal_plane_angle   d_angle_µas,
+  spec_capability          d_tag
+) RETURNS boolean AS $$
+BEGIN
+  RETURN num_nulls(spec_wavelength, spec_resolution, spec_wavelength_coverage,
+                   spec_focal_plane, spec_focal_plane_angle, spec_capability) < 6;
```

Comment:
I expected it to return `num_nulls(....) > 0` means spectroscopy mode?  Am I reading it wrong?

Reply:
you're right and it is cleaner just `>0`